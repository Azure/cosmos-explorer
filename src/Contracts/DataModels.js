"use strict";
exports.__esModule = true;
exports.SparkClusterEndpointKind = exports.ConflictResolutionMode = exports.ApiKind = void 0;
var ApiKind;
(function (ApiKind) {
    ApiKind[ApiKind["SQL"] = 0] = "SQL";
    ApiKind[ApiKind["MongoDB"] = 1] = "MongoDB";
    ApiKind[ApiKind["Table"] = 2] = "Table";
    ApiKind[ApiKind["Cassandra"] = 3] = "Cassandra";
    ApiKind[ApiKind["Graph"] = 4] = "Graph";
    ApiKind[ApiKind["MongoDBCompute"] = 5] = "MongoDBCompute";
})(ApiKind = exports.ApiKind || (exports.ApiKind = {}));
var ConflictResolutionMode;
(function (ConflictResolutionMode) {
    ConflictResolutionMode["Custom"] = "Custom";
    ConflictResolutionMode["LastWriterWins"] = "LastWriterWins";
})(ConflictResolutionMode = exports.ConflictResolutionMode || (exports.ConflictResolutionMode = {}));
var SparkClusterEndpointKind;
(function (SparkClusterEndpointKind) {
    SparkClusterEndpointKind["SparkUI"] = "SparkUI";
    SparkClusterEndpointKind["HistoryServerUI"] = "HistoryServerUI";
    SparkClusterEndpointKind["Livy"] = "Livy";
})(SparkClusterEndpointKind = exports.SparkClusterEndpointKind || (exports.SparkClusterEndpointKind = {}));
