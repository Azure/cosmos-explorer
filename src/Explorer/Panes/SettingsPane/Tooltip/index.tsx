import React, { FunctionComponent } from "react";
import InfoBubble from "../../../../../images/info-bubble.svg";

export interface TooltipProps {
  children: React.ReactNode;
}

export const Tooltip: FunctionComponent<TooltipProps> = ({ children }: TooltipProps) => {
  return (
    <span className="infoTooltip" role="tooltip" tabIndex={0}>
      <img className="infoImg" src={InfoBubble} alt="More information" />
      <span className="tooltiptext pageOptionTooltipWidth">{children}</span>
    </span>
  );
};
