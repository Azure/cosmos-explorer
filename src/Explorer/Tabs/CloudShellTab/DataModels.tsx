/**
 * Copyright (c) Microsoft Corporation.  All rights reserved.
 */

export const enum OsType {
    Linux = "linux",
    Windows = "windows"
}

export const enum ShellType {
    Bash = "bash",
    PowerShellCore = "pwsh"
}
  
export const enum NetworkType {
    Default = "Default",
    Isolated = "Isolated"
}

export const enum SessionType {
    Mounted = "Mounted",
    Ephemeral = "Ephemeral"
}

export type Settings = {
    location: string;
    sessionType: SessionType;
    osType: OsType;
};

export type ProvisionConsoleResponse = {
    properties: {
        osType: OsType;
        provisioningState: string;
        uri: string;
    };
};

export type Authorization = {
    token: string;
};

export type ConnectTerminalResponse = {
    id: string;
    idleTimeout: string;
    rootDirectory: string;
    socketUri: string;
    tokenUpdated: boolean;
};


