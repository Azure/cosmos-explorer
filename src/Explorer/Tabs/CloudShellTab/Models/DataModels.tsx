export const enum OsType {
  Linux = "linux",
  Windows = "windows",
}

export const enum ShellType {
  Bash = "bash",
  PowerShellCore = "pwsh",
}

export const enum NetworkType {
  Default = "Default",
  Isolated = "Isolated",
}

/**
 * Azure CloudShell session types:
 * - Mounted: Sessions with persistent storage via an Azure File Share mount.
 *   Files and configurations are preserved between sessions, allowing for
 *   continuity of work across multiple CloudShell sessions.
 *
 * - Ephemeral: Temporary sessions without persistent storage.
 *   All files and changes are discarded when the session ends.
 *   These sessions start faster but don't retain user data.
 *
 * The session type affects resource allocation, startup time,
 * and whether user files/configurations persist between sessions.
 */
export const enum SessionType {
  Mounted = "Mounted",
  Ephemeral = "Ephemeral",
}

export type CloudShellSettings = {
  properties: UserSettingProperties;
};

export type UserSettingProperties = {
  networkType: string;
  preferredLocation: string;
  preferredOsType: OsType;
  preferredShellType: ShellType;
  userSubscription: string;
  sessionType: SessionType;
  vnetSettings: object;
};

export type ProvisionConsoleResponse = {
  properties: {
    osType: OsType;
    provisioningState: string;
    uri: string;
  };
};

export type Authorization = {
  token: string;
};

export type ConnectTerminalResponse = {
  id: string;
  idleTimeout: string;
  rootDirectory: string;
  socketUri: string;
  tokenUpdated: boolean;
};

export type ProviderAuthorization = {
  applicationId: string;
  roleDefinitionId: string;
};

export type ProviderResourceType = {
  resourceType: string;
  locations: string[];
  apiVersions: string[];
  defaultApiVersion?: string;
  capabilities?: string;
};

export type RegistrationState = "Registered" | "NotRegistered" | "Registering" | "Unregistering";
export type RegistrationPolicy = "RegistrationRequired" | "RegistrationOptional";

export type CloudShellProviderInfo = {
  id: string;
  namespace: string;
  authorizations?: ProviderAuthorization[];
  resourceTypes: ProviderResourceType[];
  registrationState: RegistrationState;
  registrationPolicy: RegistrationPolicy;
};
