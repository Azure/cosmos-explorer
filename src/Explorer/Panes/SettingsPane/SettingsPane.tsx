import { Checkbox, ChoiceGroup, IChoiceGroupOption, SpinButton } from "@fluentui/react";
import React, { FunctionComponent, MouseEvent, useState } from "react";
import * as Constants from "../../../Common/Constants";
import { InfoTooltip } from "../../../Common/Tooltip/InfoTooltip";
import { configContext } from "../../../ConfigContext";
import { LocalStorageUtility, StorageKey } from "../../../Shared/StorageUtility";
import * as StringUtility from "../../../Shared/StringUtility";
import { userContext } from "../../../UserContext";
import { logConsoleInfo } from "../../../Utils/NotificationConsoleUtils";
import { RightPaneForm, RightPaneFormProps } from "../RightPaneForm/RightPaneForm";

export interface SettingsPaneProps {
  expandConsole: () => void;
  closePanel: () => void;
}

export const SettingsPane: FunctionComponent<SettingsPaneProps> = ({
  expandConsole,
  closePanel,
}: SettingsPaneProps) => {
  const [isExecuting, setIsExecuting] = useState<boolean>(false);
  const [pageOption, setPageOption] = useState<string>(
    LocalStorageUtility.getEntryNumber(StorageKey.ActualItemPerPage) === Constants.Queries.unlimitedItemsPerPage
      ? Constants.Queries.UnlimitedPageOption
      : Constants.Queries.CustomPageOption
  );
  const [customItemPerPage, setCustomItemPerPage] = useState<number>(
    LocalStorageUtility.getEntryNumber(StorageKey.CustomItemPerPage) || 0
  );
  const [crossPartitionQueryEnabled, setCrossPartitionQueryEnabled] = useState<boolean>(
    LocalStorageUtility.hasItem(StorageKey.IsCrossPartitionQueryEnabled)
      ? LocalStorageUtility.getEntryString(StorageKey.IsCrossPartitionQueryEnabled) === "true"
      : true
  );
  const [graphAutoVizDisabled, setGraphAutoVizDisabled] = useState<string>(
    LocalStorageUtility.hasItem(StorageKey.IsGraphAutoVizDisabled)
      ? LocalStorageUtility.getEntryString(StorageKey.IsGraphAutoVizDisabled)
      : "false"
  );
  const [maxDegreeOfParallelism, setMaxDegreeOfParallelism] = useState<number>(
    LocalStorageUtility.hasItem(StorageKey.MaxDegreeOfParellism)
      ? LocalStorageUtility.getEntryNumber(StorageKey.MaxDegreeOfParellism)
      : Constants.Queries.DefaultMaxDegreeOfParallelism
  );
  const explorerVersion = configContext.gitSha;
  const shouldShowQueryPageOptions = userContext.apiType === "SQL";
  const shouldShowGraphAutoVizOption = userContext.apiType === "Gremlin";
  const shouldShowCrossPartitionOption = userContext.apiType !== "Gremlin";
  const shouldShowParallelismOption = userContext.apiType !== "Gremlin";

  const handlerOnSubmit = (e: MouseEvent<HTMLButtonElement>) => {
    setIsExecuting(true);

    LocalStorageUtility.setEntryNumber(
      StorageKey.ActualItemPerPage,
      isCustomPageOptionSelected() ? customItemPerPage : Constants.Queries.unlimitedItemsPerPage
    );
    LocalStorageUtility.setEntryNumber(StorageKey.CustomItemPerPage, customItemPerPage);
    LocalStorageUtility.setEntryString(StorageKey.IsCrossPartitionQueryEnabled, crossPartitionQueryEnabled.toString());
    LocalStorageUtility.setEntryNumber(StorageKey.MaxDegreeOfParellism, maxDegreeOfParallelism);

    if (shouldShowGraphAutoVizOption) {
      LocalStorageUtility.setEntryBoolean(
        StorageKey.IsGraphAutoVizDisabled,
        StringUtility.toBoolean(graphAutoVizDisabled)
      );
    }

    setIsExecuting(false);
    logConsoleInfo(
      `Updated items per page setting to ${LocalStorageUtility.getEntryNumber(StorageKey.ActualItemPerPage)}`
    );
    logConsoleInfo(`${crossPartitionQueryEnabled ? "Enabled" : "Disabled"} cross-partition query feed option`);
    logConsoleInfo(
      `Updated the max degree of parallelism query feed option to ${LocalStorageUtility.getEntryNumber(
        StorageKey.MaxDegreeOfParellism
      )}`
    );

    if (shouldShowGraphAutoVizOption) {
      logConsoleInfo(
        `Graph result will be displayed as ${
          LocalStorageUtility.getEntryBoolean(StorageKey.IsGraphAutoVizDisabled) ? "JSON" : "Graph"
        }`
      );
    }

    logConsoleInfo(
      `Updated query setting to ${LocalStorageUtility.getEntryString(StorageKey.SetPartitionKeyUndefined)}`
    );
    closePanel();
    e.preventDefault();
  };

  const isCustomPageOptionSelected = () => {
    return pageOption === Constants.Queries.CustomPageOption;
  };

  const handleOnGremlinChange = (ev: React.FormEvent<HTMLInputElement>, option: IChoiceGroupOption): void => {
    setGraphAutoVizDisabled(option.key);
  };

  const genericPaneProps: RightPaneFormProps = {
    expandConsole,
    formError: "",
    isExecuting,
    submitButtonText: "Apply",
    onSubmit: () => handlerOnSubmit(undefined),
  };
  const pageOptionList: IChoiceGroupOption[] = [
    { key: Constants.Queries.CustomPageOption, text: "Custom" },
    { key: Constants.Queries.UnlimitedPageOption, text: "Unlimited" },
  ];

  const graphAutoOptionList: IChoiceGroupOption[] = [
    { key: "false", text: "Graph" },
    { key: "true", text: "JSON" },
  ];

  const handleOnPageOptionChange = (ev: React.FormEvent<HTMLInputElement>, option: IChoiceGroupOption): void => {
    setPageOption(option.key);
  };
  return (
    <RightPaneForm {...genericPaneProps}>
      <div className="paneMainContent">
        {shouldShowQueryPageOptions && (
          <div className="settingsSection">
            <div className="settingsSectionPart pageOptionsPart">
              <div className="settingsSectionLabel">
                Page options
                <InfoTooltip>
                  Choose Custom to specify a fixed amount of query results to show, or choose Unlimited to show as many
                  query results per page.
                </InfoTooltip>
              </div>
              <ChoiceGroup selectedKey={pageOption} options={pageOptionList} onChange={handleOnPageOptionChange} />
            </div>
            <div className="tabs settingsSectionPart">
              {isCustomPageOptionSelected() && (
                <div className="tabcontent">
                  <div className="settingsSectionLabel">
                    Query results per page
                    <InfoTooltip>Enter the number of query results that should be shown per page.</InfoTooltip>
                  </div>

                  <SpinButton
                    ariaLabel="Custom query items per page"
                    value={"" + customItemPerPage}
                    onIncrement={(newValue) => {
                      setCustomItemPerPage(parseInt(newValue) + 1 || customItemPerPage);
                    }}
                    onDecrement={(newValue) => setCustomItemPerPage(parseInt(newValue) - 1 || customItemPerPage)}
                    onValidate={(newValue) => setCustomItemPerPage(parseInt(newValue) || customItemPerPage)}
                    min={1}
                    step={1}
                    className="textfontclr"
                    incrementButtonAriaLabel="Increase value by 1"
                    decrementButtonAriaLabel="Decrease value by 1"
                  />
                </div>
              )}
            </div>
          </div>
        )}
        {shouldShowCrossPartitionOption && (
          <div className="settingsSection">
            <div className="settingsSectionPart">
              <div className="settingsSectionLabel">
                Enable cross-partition query
                <InfoTooltip>
                  Send more than one request while executing a query. More than one request is necessary if the query is
                  not scoped to single partition key value.
                </InfoTooltip>
              </div>

              <Checkbox
                styles={{
                  label: { padding: 0 },
                }}
                className="padding"
                ariaLabel="Enable cross partition query"
                checked={crossPartitionQueryEnabled}
                onChange={() => setCrossPartitionQueryEnabled(!crossPartitionQueryEnabled)}
              />
            </div>
          </div>
        )}
        {shouldShowParallelismOption && (
          <div className="settingsSection">
            <div className="settingsSectionPart">
              <div className="settingsSectionLabel">
                Max degree of parallelism
                <InfoTooltip>
                  Gets or sets the number of concurrent operations run client side during parallel query execution. A
                  positive property value limits the number of concurrent operations to the set value. If it is set to
                  less than 0, the system automatically decides the number of concurrent operations to run.
                </InfoTooltip>
              </div>

              <SpinButton
                min={-1}
                step={1}
                className="textfontclr"
                role="textbox"
                tabIndex={0}
                id="max-degree"
                value={"" + maxDegreeOfParallelism}
                onIncrement={(newValue) => setMaxDegreeOfParallelism(parseInt(newValue) + 1 || maxDegreeOfParallelism)}
                onDecrement={(newValue) => setMaxDegreeOfParallelism(parseInt(newValue) - 1 || maxDegreeOfParallelism)}
                onValidate={(newValue) => setMaxDegreeOfParallelism(parseInt(newValue) || maxDegreeOfParallelism)}
                ariaLabel="Max degree of parallelism"
              />
            </div>
          </div>
        )}
        {shouldShowGraphAutoVizOption && (
          <div className="settingsSection">
            <div className="settingsSectionPart">
              <div className="settingsSectionLabel">
                Display Gremlin query results as:&nbsp;
                <InfoTooltip>
                  Select Graph to automatically visualize the query results as a Graph or JSON to display the results as
                  JSON.
                </InfoTooltip>
              </div>

              <ChoiceGroup
                selectedKey={graphAutoVizDisabled}
                options={graphAutoOptionList}
                onChange={handleOnGremlinChange}
                aria-label="Graph Auto-visualization"
              />
            </div>
          </div>
        )}
        <div className="settingsSection">
          <div className="settingsSectionPart">
            <div className="settingsSectionLabel">Explorer Version</div>
            <div>{explorerVersion}</div>
          </div>
        </div>
      </div>
    </RightPaneForm>
  );
};
