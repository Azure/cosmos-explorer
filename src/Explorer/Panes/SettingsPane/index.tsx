import { Checkbox, Pivot, PivotItem, PivotLinkFormat, PivotLinkSize, SpinButton } from "office-ui-fabric-react";
import React, { FunctionComponent, MouseEvent, useEffect, useState } from "react";
import * as Constants from "../../../Common/Constants";
import { configContext } from "../../../ConfigContext";
import { LocalStorageUtility, StorageKey } from "../../../Shared/StorageUtility";
import * as StringUtility from "../../../Shared/StringUtility";
import { userContext } from "../../../UserContext";
import { logConsoleInfo } from "../../../Utils/NotificationConsoleUtils";
import Explorer from "../../Explorer";
import { GenericRightPaneComponent, GenericRightPaneProps } from "../GenericRightPaneComponent";
import { Tooltip } from "./Tooltip";

export interface SettingsPaneProps {
  explorer: Explorer;
  closePanel: () => void;
}

export const SettingsPane: FunctionComponent<SettingsPaneProps> = ({
  explorer: container,
  closePanel,
}: SettingsPaneProps) => {
  const [formErrors, setFormErrors] = useState<string>("");
  const [formErrorsDetails, setFormErrorsDetails] = useState<string>("");
  const [isExecuting, setIsExecuting] = useState<boolean>(false);
  const [pageOption, setPageOption] = useState<string>("");
  const [customItemPerPage, setCustomItemPerPage] = useState<number>(0);
  const [crossPartitionQueryEnabled, setCrossPartitionQueryEnabled] = useState<boolean>(
    LocalStorageUtility.hasItem(StorageKey.IsCrossPartitionQueryEnabled)
      ? LocalStorageUtility.getEntryString(StorageKey.IsCrossPartitionQueryEnabled) === "true"
      : true
  );
  const [graphAutoVizDisabled, setGraphAutoVizDisabled] = useState<string>(
    LocalStorageUtility.hasItem(StorageKey.IsGraphAutoVizDisabled)
      ? LocalStorageUtility.getEntryString(StorageKey.IsGraphAutoVizDisabled)
      : "false"
  );
  const [maxDegreeOfParallelism, setMaxDegreeOfParallelism] = useState<number>(
    LocalStorageUtility.hasItem(StorageKey.MaxDegreeOfParellism)
      ? LocalStorageUtility.getEntryNumber(StorageKey.MaxDegreeOfParellism)
      : Constants.Queries.DefaultMaxDegreeOfParallelism
  );
  const explorerVersion = configContext.gitSha;
  const shouldShowQueryPageOptions = userContext.apiType === "SQL";
  const shouldShowGraphAutoVizOption = userContext.apiType === "Gremlin";
  const shouldShowCrossPartitionOption = userContext.apiType !== "Gremlin";
  const shouldShowParallelismOption = userContext.apiType !== "Gremlin";

  useEffect(() => {
    _loadSettings();
  }, []);

  const handlerOnSubmit = (e: MouseEvent<HTMLButtonElement>) => {
    setFormErrors("");
    setIsExecuting(true);

    LocalStorageUtility.setEntryNumber(
      StorageKey.ActualItemPerPage,
      isCustomPageOptionSelected() ? customItemPerPage : Constants.Queries.unlimitedItemsPerPage
    );
    LocalStorageUtility.setEntryNumber(StorageKey.CustomItemPerPage, customItemPerPage);
    LocalStorageUtility.setEntryString(StorageKey.IsCrossPartitionQueryEnabled, crossPartitionQueryEnabled.toString());
    LocalStorageUtility.setEntryNumber(StorageKey.MaxDegreeOfParellism, maxDegreeOfParallelism);

    if (shouldShowGraphAutoVizOption) {
      LocalStorageUtility.setEntryBoolean(
        StorageKey.IsGraphAutoVizDisabled,
        StringUtility.toBoolean(graphAutoVizDisabled)
      );
    }

    setIsExecuting(false);
    logConsoleInfo(
      `Updated items per page setting to ${LocalStorageUtility.getEntryNumber(StorageKey.ActualItemPerPage)}`
    );
    logConsoleInfo(`${crossPartitionQueryEnabled ? "Enabled" : "Disabled"} cross-partition query feed option`);
    logConsoleInfo(
      `Updated the max degree of parallelism query feed option to ${LocalStorageUtility.getEntryNumber(
        StorageKey.MaxDegreeOfParellism
      )}`
    );

    if (shouldShowGraphAutoVizOption) {
      logConsoleInfo(
        `Graph result will be displayed as ${
          LocalStorageUtility.getEntryBoolean(StorageKey.IsGraphAutoVizDisabled) ? "JSON" : "Graph"
        }`
      );
    }

    logConsoleInfo(
      `Updated query setting to ${LocalStorageUtility.getEntryString(StorageKey.SetPartitionKeyUndefined)}`
    );

    closePanel();

    e.preventDefault();
  };

  const isCustomPageOptionSelected = () => {
    return pageOption === Constants.Queries.CustomPageOption;
  };

  const _loadSettings = () => {
    setIsExecuting(true);
    try {
      setPageOption(
        LocalStorageUtility.getEntryNumber(StorageKey.ActualItemPerPage) === Constants.Queries.unlimitedItemsPerPage
          ? Constants.Queries.UnlimitedPageOption
          : Constants.Queries.CustomPageOption
      );
      setPageOption(
        LocalStorageUtility.getEntryNumber(StorageKey.ActualItemPerPage) === Constants.Queries.unlimitedItemsPerPage
          ? Constants.Queries.UnlimitedPageOption
          : Constants.Queries.CustomPageOption
      );
      setCustomItemPerPage(LocalStorageUtility.getEntryNumber(StorageKey.CustomItemPerPage));
    } catch (exception) {
      setFormErrors("Unable to load your settings");
      setFormErrorsDetails(exception);
    } finally {
      setIsExecuting(false);
    }
  };

  const onPivotChange = (item: PivotItem): void => {
    setPageOption(item.props.itemKey);
  };
  const onGremlinPivotChange = (item: PivotItem): void => {
    setGraphAutoVizDisabled(item.props.itemKey);
  };
  const genericPaneProps: GenericRightPaneProps = {
    container,
    formError: formErrors,
    formErrorDetail: formErrorsDetails,
    id: "settingspane",
    isExecuting,
    title: "Setting",
    submitButtonText: "Apply",
    onClose: () => closePanel(),
    onSubmit: () => handlerOnSubmit(undefined),
  };

  return (
    <GenericRightPaneComponent {...genericPaneProps}>
      <div className="paneMainContent">
        <div>
          {shouldShowQueryPageOptions && (
            <div className="settingsSection">
              <div className="settingsSectionPart pageOptionsPart">
                <div className="settingsSectionLabel">
                  Page options
                  <Tooltip>
                    Choose Custom to specify a fixed amount of query results to show, or choose Unlimited to show as
                    many query results per page.
                  </Tooltip>
                </div>

                <Pivot
                  aria-label="Page options"
                  linkSize={PivotLinkSize.normal}
                  linkFormat={PivotLinkFormat.tabs}
                  onLinkClick={onPivotChange}
                  selectedKey={pageOption}
                >
                  <PivotItem headerText="Custom" itemKey={Constants.Queries.CustomPageOption}></PivotItem>
                  <PivotItem headerText="Unlimited" itemKey={Constants.Queries.UnlimitedPageOption}></PivotItem>
                </Pivot>
              </div>
              <div className="tabs settingsSectionPart">
                {isCustomPageOptionSelected() && (
                  <div className="tabcontent">
                    <div className="settingsSectionLabel">
                      Query results per page
                      <Tooltip>Enter the number of query results that should be shown per page.</Tooltip>
                    </div>

                    <SpinButton
                      ariaLabel="Custom query items per page"
                      value={"" + customItemPerPage}
                      onIncrement={(newValue) => {
                        setCustomItemPerPage(parseInt(newValue) + 1 || customItemPerPage);
                      }}
                      onDecrement={(newValue) => setCustomItemPerPage(parseInt(newValue) - 1 || customItemPerPage)}
                      min={1}
                      step={1}
                      className="textfontclr"
                      incrementButtonAriaLabel="Increase value by 1"
                      decrementButtonAriaLabel="Decrease value by 1"
                    />
                  </div>
                )}
              </div>
            </div>
          )}
          {shouldShowCrossPartitionOption && (
            <div className="settingsSection">
              <div className="settingsSectionPart">
                <div className="settingsSectionLabel">
                  Enable cross-partition query
                  <Tooltip>
                    Send more than one request while executing a query. More than one request is necessary if the query
                    is not scoped to single partition key value.
                  </Tooltip>
                </div>

                <Checkbox
                  style={{ padding: "0" }}
                  className="padding"
                  tabIndex={0}
                  ariaLabel="Enable cross partition query"
                  checked={crossPartitionQueryEnabled}
                  onChange={() => setCrossPartitionQueryEnabled(!crossPartitionQueryEnabled)}
                />
              </div>
            </div>
          )}
          {shouldShowParallelismOption && (
            <div className="settingsSection">
              <div className="settingsSectionPart">
                <div className="settingsSectionLabel">
                  Max degree of parallelism
                  <Tooltip>
                    Gets or sets the number of concurrent operations run client side during parallel query execution. A
                    positive property value limits the number of concurrent operations to the set value. If it is set to
                    less than 0, the system automatically decides the number of concurrent operations to run.
                  </Tooltip>
                </div>

                <SpinButton
                  min={-1}
                  step={1}
                  className="textfontclr"
                  role="textbox"
                  tabIndex={0}
                  id="max-degree"
                  value={"" + maxDegreeOfParallelism}
                  onIncrement={(newValue) =>
                    setMaxDegreeOfParallelism(parseInt(newValue) + 1 || maxDegreeOfParallelism)
                  }
                  onDecrement={(newValue) =>
                    setMaxDegreeOfParallelism(parseInt(newValue) - 1 || maxDegreeOfParallelism)
                  }
                  ariaLabel="Max degree of parallelism"
                />
              </div>
            </div>
          )}
          {shouldShowGraphAutoVizOption && (
            <div className="settingsSection">
              <div className="settingsSectionPart">
                <div className="settingsSectionLabel">
                  Display Gremlin query results as:&nbsp;
                  <Tooltip>
                    Select Graph to automatically visualize the query results as a Graph or JSON to display the results
                    as JSON.
                  </Tooltip>
                </div>

                <Pivot
                  aria-label="Graph Auto-visualization"
                  linkSize={PivotLinkSize.normal}
                  linkFormat={PivotLinkFormat.tabs}
                  onLinkClick={onGremlinPivotChange}
                  selectedKey={graphAutoVizDisabled}
                >
                  <PivotItem headerText="Graph" itemKey="false"></PivotItem>
                  <PivotItem headerText="JSON" itemKey="true"></PivotItem>
                </Pivot>
              </div>
            </div>
          )}
          <div className="settingsSection">
            <div className="settingsSectionPart">
              <div className="settingsSectionLabel">Explorer Version</div>
              <div>{explorerVersion}</div>
            </div>
          </div>
        </div>
      </div>
    </GenericRightPaneComponent>
  );
};
