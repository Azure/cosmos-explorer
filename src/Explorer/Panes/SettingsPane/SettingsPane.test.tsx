import { shallow } from "enzyme";
import React from "react";
import { DatabaseAccount } from "../../../Contracts/DataModels";
import { updateUserContext } from "../../../UserContext";
import { SettingsPane } from "./SettingsPane";

describe("Settings Pane", () => {
  it("should render Default properly", () => {
    const wrapper = shallow(<SettingsPane explorer={null} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render Gremlin properly", () => {
    updateUserContext({
      databaseAccount: {
        properties: {
          capabilities: [{ name: "EnableGremlin" }],
        },
      } as DatabaseAccount,
    });
    const wrapper = shallow(<SettingsPane explorer={null} />);
    expect(wrapper).toMatchSnapshot();
  });
});
