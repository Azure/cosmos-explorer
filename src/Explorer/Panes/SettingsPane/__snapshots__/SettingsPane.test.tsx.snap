// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Pane should render Default properly 1`] = `
<GenericRightPaneComponent
  expandConsole={[Function]}
  formError=""
  formErrorDetail=""
  id="settingspane"
  isExecuting={false}
  onClose={[Function]}
  onSubmit={[Function]}
  submitButtonText="Apply"
  title="Setting"
>
  <div
    className="paneMainContent"
  >
    <div
      className="settingsSection"
    >
      <div
        className="settingsSectionPart pageOptionsPart"
      >
        <div
          className="settingsSectionLabel"
        >
          Page options
          <Tooltip>
            Choose Custom to specify a fixed amount of query results to show, or choose Unlimited to show as many query results per page.
          </Tooltip>
        </div>
        <StyledChoiceGroupBase
          onChange={[Function]}
          options={
            Array [
              Object {
                "key": "custom",
                "text": "Custom",
              },
              Object {
                "key": "unlimited",
                "text": "Unlimited",
              },
            ]
          }
          selectedKey="custom"
        />
      </div>
      <div
        className="tabs settingsSectionPart"
      >
        <div
          className="tabcontent"
        >
          <div
            className="settingsSectionLabel"
          >
            Query results per page
            <Tooltip>
              Enter the number of query results that should be shown per page.
            </Tooltip>
          </div>
          <CustomizedSpinButton
            ariaLabel="Custom query items per page"
            className="textfontclr"
            decrementButtonAriaLabel="Decrease value by 1"
            decrementButtonIcon={
              Object {
                "iconName": "ChevronDownSmall",
              }
            }
            disabled={false}
            incrementButtonAriaLabel="Increase value by 1"
            incrementButtonIcon={
              Object {
                "iconName": "ChevronUpSmall",
              }
            }
            label=""
            labelPosition={2}
            max={100}
            min={1}
            onDecrement={[Function]}
            onIncrement={[Function]}
            onValidate={[Function]}
            step={1}
            value="0"
          />
        </div>
      </div>
    </div>
    <div
      className="settingsSection"
    >
      <div
        className="settingsSectionPart"
      >
        <div
          className="settingsSectionLabel"
        >
          Enable cross-partition query
          <Tooltip>
            Send more than one request while executing a query. More than one request is necessary if the query is not scoped to single partition key value.
          </Tooltip>
        </div>
        <StyledCheckboxBase
          ariaLabel="Enable cross partition query"
          checked={true}
          className="padding"
          onChange={[Function]}
          styles={
            Object {
              "label": Object {
                "padding": 0,
              },
            }
          }
          tabIndex={0}
        />
      </div>
    </div>
    <div
      className="settingsSection"
    >
      <div
        className="settingsSectionPart"
      >
        <div
          className="settingsSectionLabel"
        >
          Max degree of parallelism
          <Tooltip>
            Gets or sets the number of concurrent operations run client side during parallel query execution. A positive property value limits the number of concurrent operations to the set value. If it is set to less than 0, the system automatically decides the number of concurrent operations to run.
          </Tooltip>
        </div>
        <CustomizedSpinButton
          ariaLabel="Max degree of parallelism"
          className="textfontclr"
          decrementButtonIcon={
            Object {
              "iconName": "ChevronDownSmall",
            }
          }
          disabled={false}
          id="max-degree"
          incrementButtonIcon={
            Object {
              "iconName": "ChevronUpSmall",
            }
          }
          label=""
          labelPosition={2}
          max={100}
          min={-1}
          onDecrement={[Function]}
          onIncrement={[Function]}
          onValidate={[Function]}
          role="textbox"
          step={1}
          tabIndex={0}
          value="6"
        />
      </div>
    </div>
    <div
      className="settingsSection"
    >
      <div
        className="settingsSectionPart"
      >
        <div
          className="settingsSectionLabel"
        >
          Explorer Version
        </div>
        <div />
      </div>
    </div>
  </div>
</GenericRightPaneComponent>
`;

exports[`Settings Pane should render Gremlin properly 1`] = `
<GenericRightPaneComponent
  expandConsole={[Function]}
  formError=""
  formErrorDetail=""
  id="settingspane"
  isExecuting={false}
  onClose={[Function]}
  onSubmit={[Function]}
  submitButtonText="Apply"
  title="Setting"
>
  <div
    className="paneMainContent"
  >
    <div
      className="settingsSection"
    >
      <div
        className="settingsSectionPart"
      >
        <div
          className="settingsSectionLabel"
        >
          Display Gremlin query results as:Â 
          <Tooltip>
            Select Graph to automatically visualize the query results as a Graph or JSON to display the results as JSON.
          </Tooltip>
        </div>
        <StyledChoiceGroupBase
          aria-label="Graph Auto-visualization"
          onChange={[Function]}
          options={
            Array [
              Object {
                "key": "false",
                "text": "Graph",
              },
              Object {
                "key": "true",
                "text": "JSON",
              },
            ]
          }
          selectedKey="false"
        />
      </div>
    </div>
    <div
      className="settingsSection"
    >
      <div
        className="settingsSectionPart"
      >
        <div
          className="settingsSectionLabel"
        >
          Explorer Version
        </div>
        <div />
      </div>
    </div>
  </div>
</GenericRightPaneComponent>
`;
